### ISSUE TITLE
gradient_descent (optsr.py) crashes with more than 2 types of parameters

### ISSUE THREAD

**Issue by lkwagner** [WagnerGroup/pyqmc#50]
Here is some code that exhibits the problem: 
```python
from pyscf import gto,scf 
import pyqmc

r=1.54/.529177
mol = gto.M(atom='H 0. 0. 0.; H 0. 0. %g'%r, ecp='bfd',basis='bfd_vtz',unit='bohr',verbose=1)
mf=scf.RHF(mol).run()
wf=pyqmc.slater_jastrow(mol,mf)
enacc=pyqmc.EnergyAccumulator(mol)
print(list(wf.parameters.keys()))

nconfig=500
coords=pyqmc.initial_guess(mol,nconfig)
wf,graddata=pyqmc.gradient_descent(wf,coords,warmup=1,
        params=['wf1mo_coeff','wf2acoeff','wf2bcoeff'],
                             accumulators={'pgrad':pyqmc.PGradAccumulator(enacc)},
                             step=0.1,maxiters=10)
```

**Comment by lkwagner** [comment]
I think that this could be fixed at the same time as #44, with the concept of a transform; which takes in the gradient dictionary and outputs a linearization. I think that PGradAccumulator should use the same transform.

**Comment by lkwagner** [comment]
Closed by recent commit.
